### PR TITLE
Split extension manager out to separate module

### PR DESCRIPTION
**Pull request by brandones**
https://github.com/openmrs/esm-extension-manager/pull/1

### CODE REVIEW

File: spa/package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "systemjs": "^6.3.3"
   },
   "devDependencies": {
+    "@openmrs/esm-extension-manager": "file:../../mf/openmrs-esm-extension-manager",
```
Comment (brandones) [inline spa/package.json]: This must be replaced by the actual package once the package is published (before this PR is committed)

File: spa/src/index.ts [hunk 2]
```diff
@@ -78,33 +76,27 @@ function preprocessActivator(
...
     if (typeof setup === "function") {
       const result = setup();
 
       if (result && typeof result === "object") {
-        System.import(singleSpa).then(({ registerApplication }) => {
-          const availableExtensions: Array<ExtensionDefinition> =
-            result.extensions ?? [];
+        System.import("single-spa").then(({ registerApplication }) => {
+          System.import("@openmrs/esm-extension-manager").then(({ registerExtension }) => {
+            const availableExtensions = result.extensions ?? [];
```
Comment (denniskigen) [inline spa/src/index.ts]: Should we keep the type annotations for `availableExtensions`? `Array<ExtensionDefinition>`

Comment (brandones) [reply spa/src/index.ts]: Yeah actually, now that I think of it, we can import that type statically. I'll add that back in.

**Review by jonathandick — APPROVED** [review]
LGTM

**Pull request merged**